rcl/include/rcl/service.h

```diff
@@ -0,0 +1,318 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__SERVICE_H_
+#define RCL__SERVICE_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "rosidl_generator_c/service_type_support.h"
+
+#include "rcl/macros.h"
+#include "rcl/node.h"
+#include "rcl/visibility_control.h"
+
+/// Internal rcl implementation struct.
+struct rcl_service_impl_t;
+
+/// Handle for a rcl service.
+typedef struct rcl_service_t
+{
+  struct rcl_service_impl_t * impl;
+} rcl_service_t;
+
+/// Options available for a rcl service.
+typedef struct rcl_service_options_t
+{
+  /// Middleware quality of service settings for the service.
+  rmw_qos_profile_t qos;
+  /// Custom allocator for the service, used for incidental allocations.
+  /* For default behavior (malloc/free), see: rcl_get_default_allocator() */
+  rcl_allocator_t allocator;
+} rcl_service_options_t;
+
+/// Return a rcl_service_t struct with members set to NULL.
+/* Should be called to get a null rcl_service_t before passing to
+ * rcl_initalize_service().
+ * It's also possible to use calloc() instead of this if the rcl_service_t
+ * is being allocated on the heap.
+ */
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_service_t
+rcl_get_zero_initialized_service(void);
+
+/// Initialize a ROS service.
```

Comment:
In `client.h` the equivalent line is something like "Initialize a rcl client". I don't have a preference for one over the other, but I think they should be consistent.